### PR TITLE
course: task 2.2 ajax

### PR DESCRIPTION
**Pull request by BorislavAnchev**
Working 2.2 AJAX task with content fetched from the
Node server.

### CODE REVIEW

File: 22AJAX/zad2.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <p id="content">Show resource content.</p>
+      </div>
+     
+      <script>
+        const loadDoc = () => {
+          const xhttp = new XMLHttpRequest();;
+          xhttp.onreadystatechange = function() {
+            if (this.readyState == 4 && this.status == 200) {
+              const responseObject = JSON.parse(this.response);
+              console.log(responseObject);
```
Comment (mgenov) [inline 22AJAX/zad2.html]: console.log could be removed

File: 22AJAX/zad2.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+              console.log('Type: ', typeof responseObject);
+              document.getElementById("content").innerHTML = responseObject.map(element => element.iban + '<br>');
+            }
+          };
+          xhttp.open("GET", "http://localhost:3001/accounts", true);
```
Comment (mgenov) [inline 22AJAX/zad2.html]: using of hard-coded host and port doesn't look reasonable as these values could be changed depending on the environment.

File: RESTful-API/api/models/accounts.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
```
Comment (mgenov) [inline RESTful-API/api/models/accounts.js]: You can remove the mongodb part as it's not required for this task.

File: RESTful-API/app.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  useNewUrlParser: true,
+  useUnifiedTopology: true
+});
+
+app.use(morgan('dev'));
+app.use(bodyParser.urlencoded({extended: false }));
+app.use(bodyParser.json());
+
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
```
Comment (mgenov) [inline RESTful-API/app.js]: I'm not sure that you need these headers for this task.

File: RESTful-API/app.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  res.header('Access-Control-Allow-Headers',
+  'Origin, X-Requested-With, Content-Type, Accept, Authorization'
+  );
+  if(req.method === 'OPTIONS') {
```
Comment (mgenov) [inline RESTful-API/app.js]: Options method is not used in this task.

File: 22AJAX/zad2.html (hunk 1 above)
Comment (BorislavAnchev) [reply 22AJAX/zad2.html]: Removed.

File: 22AJAX/zad2.html (hunk 2 above)
Comment (BorislavAnchev) [reply 22AJAX/zad2.html]: How should I do it then?

Comment (mgenov) [reply 22AJAX/zad2.html]: Relative paths should work as expected.

File: RESTful-API/api/models/accounts.js (hunk 3 above)
Comment (BorislavAnchev) [reply RESTful-API/api/models/accounts.js]: Removed.

**Comment by BorislavAnchev** [comment]
New code ready for review.

**Comment by BorislavAnchev** [comment]
New code ready for review.

**Pull request merged**